lib/git.ncl

```diff
@@ -0,0 +1,52 @@
+# let nix = import "./nix-interop/nix.ncl" in
+let filegen = import "./files.ncl" in
+let Gitignorable = {
+  gitignore
+    : Bool
+    | doc "Whether to gitignore the file"
+    | default
+    = false,
+  ..
+}
+in
+let header : String = m%"
+    ##########
+    # This file is autogenerated by Organist, don't edit
+    ##########
+
+  "%
+  in
+let ignored_files | { _ : Gitignorable } -> { _ : Bool } = fun files =>
+    files
+    |> std.record.to_array
+    |> std.array.filter (fun { value = { gitignore, .. }, .. } => gitignore)
+    |> std.array.map (fun { value = { target, .. }, .. } => { field = target, value = true })
+    |> std.record.from_array
+  in
+let gitignore_content | { _ : Bool } -> String = fun ignores =>
```

Comment:
I think these functions should be in the final record so that users can add more `.gitignore` files to their repos.